apps/rush-lib/src/api/CommandLineJson.ts

```diff
@@ -31,12 +31,19 @@ export interface IBulkCommandJson extends IBaseCommandJson {
 }
 
 /**
- * "phasedCommand" from command-line.schema.json
+ * Base interface shared by the "phasedCommand" JSON entries and the post-processed
+ * "IPhase" interface in the CommandLineConfiguration
  */
-export interface IPhasedCommandJson extends IBaseCommandJson {
+export interface IPhasedCommandWithoutPhasesJson extends IBaseCommandJson {
```

Comment:
I still think this would be cleaner with a generic type on `phases` and `skipPhasesForCommand`

Reply:
It's a bit tricky to use a generic to discard a property from an object; `skipPhasesForCommand` gets trimmed during resolution.

Reply:
That would make this:
```ts
export interface IPhasedCommandJson<TPhases = string[]> extends IBaseCommandJson {
  ...
  phases: TPhases;
  skipPhasesForCommand?: TPhases extends string[] ? TPhases : never;
}
```
Which I think is generally less readable.